plugins/woocommerce-admin/client/inbox-panel/test/index.js

```diff
@@ -114,10 +149,67 @@ describe( 'inbox_panel_view_event', () => {
 			notesHaveResolved: true,
 			isBatchUpdating: false,
 		} ) );
+		InboxNoteCard.mockImplementation( ( { note } ) => {
+			return <div>{ note.id }</div>;
+		} );
 		render( <InboxPanel /> );
 
 		expect( recordEvent ).toHaveBeenCalledWith( 'inbox_panel_view', {
 			total: 5,
 		} );
 	} );
 } );
+
+describe( 'inbox_note_view event', () => {
+	test( 'should fire when inbox note card calls onNoteVisible', () => {
+		notes.forEach( ( note ) => ( note.is_deleted = false ) );
+		useSelect.mockImplementation( () => ( {
+			notes,
+			isError: false,
+			notesHaveResolved: true,
+			isBatchUpdating: false,
+		} ) );
+		InboxNoteCard.mockImplementation( ( { onNoteVisible, note } ) => {
+			useEffect( () => onNoteVisible( note ), [] );
+			return <div>{ note.id }</div>;
+		} );
+		render( <InboxPanel /> );
+		notes.forEach( ( note ) => {
+			expect( recordEvent ).toHaveBeenCalledWith( 'inbox_note_view', {
+				note_content: note.content,
+				note_name: note.name,
+				note_title: note.title,
+				note_type: note.type,
+				screen: '',
+			} );
+		} );
+	} );
+} );
+
+describe( 'inbox_action_click event', () => {
+	test( 'should fire tracks event when inboxNoteCard fires onBodyLinkClick', () => {
+		useSelect.mockImplementation( () => ( {
+			notes,
+			isError: false,
+			notesHaveResolved: true,
+			isBatchUpdating: false,
+		} ) );
+		InboxNoteCard.mockImplementation( ( { onBodyLinkClick, note } ) => {
+			return (
+				<button onClick={ () => onBodyLinkClick( note, 'innerLink' ) }>
+					Trigger action
+				</button>
+			);
+		} );
+		const { getAllByText } = render( <InboxPanel /> );
+		const buttons = getAllByText( 'Trigger action' );
+		buttons.forEach( ( button ) => userEvent.click( button ) );
+		notes.forEach( ( note ) => {
+			expect( recordEvent ).toHaveBeenCalledWith( 'inbox_action_click', {
+				note_name: note.name,
+				note_title: note.title,
+				note_content_inner_link: 'innerLink',
+			} );
+		} );
```

Comment:
Same question as above... are we really testing anything more by checking each note vs just checking the first/last/random one?